spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractReplyProducingMessageHandler.java

```diff
@@ -138,13 +138,37 @@ protected Object doInvokeAdvisedRequestHandler(Message<?> message) {
 	protected abstract Object handleRequestMessage(Message<?> requestMessage);
 
 
+	/**
+	 * An implementation of this interface is used to wrap the
+	 * {@link AbstractReplyProducingMessageHandler#handleRequestMessage(Message)}
+	 * method. Also allows access to the underlying
+	 * {@link AbstractReplyProducingMessageHandler} to obtain properties.
+	 *
+	 * @author Gary Russell
+	 * @since 2.2
+	 *
+	 * @see #getAdvisedHandler()
+	 *
+	 */
 	public interface RequestHandler {
 
 		Object handleRequestMessage(Message<?> requestMessage);
 
 		@Override
 		String toString();
 
+		/**
+		 * Utility method, intended for use in message handler advice classes to get
+		 * information about the advised object. For example:
+		 * <p>
+		 * {@code ((AbstractReplyProducingMessageHandler.RequestHandler)
+		 * invocation.getThis()).getAdvisedHandler().getComponentName()}
+		 * @return the outer class instance.
+		 *
+		 * @since 4.3.2
```

Comment:
I know what confused me else.
You have a
> cherry-pick to 4.2.x

phrase in the PR header.

But I agree with the original premise that a backport may be postponed for demand.
WDYT?